### ISSUE TITLE
The Ability to Cancel 

### ISSUE THREAD

**Issue by chalkers** [thingsSDK/flasher.js#5]
If you select the wrong port, we need to abort.

Either...RomComm could automatically do it. It needs to know it's talking to the wrong thing and throws an error for the Promise can handle it...other wise Promises aren't cancelable :(

Also given the end user the ability to cancel mid process would be ideal.

**Comment by chalkers** [comment]
@craigsdennis Once we've ripped out the promises this is possible :)

**Comment by chalkers** [comment]
Hey @craigsdennis can we cancel with rom-comm now?

**Comment by craigsdennis** [comment]
Indeed!  Let's talk about what that API should look like, but it is definitely possible to deliver a poison pill.  State will be whackadoo though.
